src/token/helpers.ts

```diff
@@ -0,0 +1,233 @@
+import type { BufferList } from 'bl/BufferList';
+
+export type Result<T> = {
+  value: T;
+  offset: number;
+};
+
+export class NotEnoughDataError extends Error {
```

Comment:
I don't think this actually needs to inherit from `Error`, and I can imagine this being expensive when there's a lot of these errors (e.g. when packetsize is low). We can change this later to remove the inheritance, because we don't actually rely on things like the location of where this error is created being recorded.